### PR TITLE
add document about stale read transaction

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

Add document about stale read transaction.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6347, https://github.com/pingcap/docs-cn/pull/6453 https://github.com/pingcap/docs-cn/pull/6476
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: best-practices/three-dc-local-read.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# Local Read under Three Data Centers Deployment
+
+In the model of three data centers, a Region has three replicas which are isolated in each data center. However, due to the requirement of strongly consistent read, TiDB must access the Leader replica of the corresponding data for every query. If the query is generated in a data center different from that of the Leader replica, TiDB needs to read data from another data center, thus causing the access latency to increase.
+
+This document describes how to use the [Stale Read](/stale-read.md) feature to avoid cross-center access and reduce the access latency at the expense of real-time data availability.
+
+## Deploy a TiDB cluster of three data centers
+
+For how to deploy three data centers, refer to [Multiple Data Centers in One City Deployment](/multi-data-centers-in-one-city-deployment.md)
```
Comment (TomShawn) [inline best-practices/three-dc-local-read.md:14]:
```suggestion
For the three-data-center deployment method, refer to [Multiple Data Centers in One City Deployment](/multi-data-centers-in-one-city-deployment.md).
```

File: best-practices/three-dc-local-read.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Note that if both the TiKV and TiDB nodes have the configuration item `labels` configured, the TiKV and TiDB nodes in the same data center must have the same value for the `zone` label. For example, if a TiKV node and a TiDB node are both in the data center `dc-1`, then the two nodes need to be configured with the following label:
+
+```
+[labels]
+zone=dc-1
+```
+
+## Perform local read using stale reads
```
Comment (TomShawn) [inline best-practices/three-dc-local-read.md:23]:
```suggestion
## Perform local read using Stale Read
```

File: best-practices/three-dc-local-read.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+[Stale Read](/stale-read.md) is a mechanism that TiDB provides for the users to read historical data. With this mechanism, you can read the corresponding historical data of a specific point in time or within a specified time range, and thus save the latency brought by data replication between storage nodes. When using stale reads in some scenarios of geo-distributed deployment, TiDB sacrifices some real-time performance but accesses the replica in the current data center to read the corresponding data, which avoids network latency brought by cross-center connection and reduces the access latency for the entire query process.
```
Comment (TomShawn) [inline best-practices/three-dc-local-read.md:25]:
```suggestion
[Stale Read](/stale-read.md) is a mechanism that TiDB provides for the users to read historical data. Using this mechanism, you can read the corresponding historical data of a specific point in time or within a specified time range, and thus save the latency brought by data replication between storage nodes. When using Stale Read in some scenarios of geo-distributed deployment, TiDB accesses the replica in the current data center to read the corresponding data at the expense of some real-time performance, which avoids network latency brought by cross-center connection and reduces the access latency for the entire query process.
```

File: best-practices/three-dc-local-read.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+When TiDB receives a stale read query, if the `zone` label of that TiDB node is configured, then TiDB sends the request to the TiKV node with the same `zone` label where the corresponding data replica resides.
```
Comment (TomShawn) [inline best-practices/three-dc-local-read.md:27]:
```suggestion
When TiDB receives a Stale Read query, if the `zone` label of that TiDB node is configured, then TiDB sends the request to the TiKV node with the same `zone` label where the corresponding data replica resides.
```

File: best-practices/three-dc-local-read.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+For how to perform stale reads, see [Read historical data using the `AS OF TIMESTAMP` clause](/as-of-timestamp.md).
```
Comment (TomShawn) [inline best-practices/three-dc-local-read.md:29]:
```suggestion
For how to perform Stale Read, see [Perform Stale Read using the `AS OF TIMESTAMP` clause](/as-of-timestamp.md).
```

File: stale-read.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+---
+title: Usage Scenarios of Stale Read
+summary: Learn about Stale Read and its usage scenarios.
+---
+
+# Usage Scenarios of Stale Read
+
+This document describes the usage scenarios of stale read. Stale read is a mechanism that TiDB applies to read historical versions of data stored in TiDB. With this mechanism, you can read the corresponding historical data of a specific point in time or within a specified time range, and thus save the latency brought by data replication between storage nodes.
+
+Internally, stale read allows TiDB to read from any replica the data of the specified point in time or the data as new as possible within the specified time range, and to always ensure the data consistency constraint during the reading process.
```
Comment (TomShawn) [inline stale-read.md:10]:
```suggestion
In terms of the internal implementation, Stale Read allows TiDB to read from any replica the data of the specified point in time or the data as new as possible within the specified time range, and to always ensure the data consistency constraint during the reading process.
```

File: stale-read.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+## Scenario examples
+
++ Scenario one: If a transaction only involves read operations and is tolerant of data staleness to some extent, you can use stale reads to get historical data. With stale read, by sacrificing some real-time performance, TiDB sends the query requests to any replica of the corresponding data and thus increases the throughput of query executions.
```
Comment (TomShawn) [inline stale-read.md:14]:
```suggestion
+ Scenario one: If a transaction only involves read operations and is tolerant of data staleness to some extent, you can use Stale Read to get historical data. Using Stale Read, TiDB sends the query requests to any replica of the corresponding data at the expense of some real-time performance, and thus increases the throughput of query executions.
```

File: stale-read.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
++ Scenario two: In some scenarios where small tables are queried, if strongly consistent reads are used, data may be concentrated on a certain storage node, causing the query pressure to be concentrated on that node as well. Therefore, the node may become a bottleneck for the whole query. With stale read, TiDB distributes the query requests to each replica of the corresponding data, which can improve the overall query throughput and significantly improve the query performance.
```
Comment (TomShawn) [inline stale-read.md:16]:
```suggestion
+ Scenario two: In some scenarios where small tables are queried, if strongly consistent reads are used, data might be concentrated on a certain storage node, causing the query pressure to be concentrated on that node as well. Therefore, that node might become a bottleneck for the whole query. With Stale Read, TiDB distributes the query requests to each replica of the corresponding data, which can improve the overall query throughput and significantly improve the query performance.
```

File: stale-read.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
++ Scenario three: In some scenarios of geo-distributed deployment, if strongly consistent follower reads are used, in order to make sure that the data read from the Follower is consistent with that stored in the Leader, TiDB requests `Readindex` from different data centers for verification, which increases the access latency for the whole query process. With stale read, by sacrificing some real-time performance, TiDB accesses the replica in the current data center to read the corresponding data, which avoids network latency brought by cross-center connection and reduces the access latency for the entire query. For more information, see [Local Read under Three Data Centers Deployment](/best-practices/three-dc-local-read.md).
```
Comment (TomShawn) [inline stale-read.md:18]:
```suggestion
+ Scenario three: In some scenarios of geo-distributed deployment, if strongly consistent follower reads are used, to make sure that the data read from the Followers is consistent with that stored in the Leader, TiDB requests `Readindex` from different data centers for verification, which increases the access latency for the whole query process. With Stale Read, TiDB accesses the replica in the current data center to read the corresponding data at the expense of some real-time performance, which avoids network latency brought by cross-center connection and reduces the access latency for the entire query. For more information, see [Local Read under Three Data Centers Deployment](/best-practices/three-dc-local-read.md).
```

File: as-of-timestamp.md [hunk 10]
```diff
@@ -0,0 +1,261 @@
+---
+title: Read Historical Data Using the `AS OF TIMESTAMP` Clause
```
Comment (TomShawn) [inline as-of-timestamp.md:2]:
```suggestion
title: Perform Stale Read Using the `AS OF TIMESTAMP` Clause
```

File: as-of-timestamp.md [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+
+```sql
+commit;
+```
+
+```
+Query OK, 0 rows affected (0.00 sec)
+```
+
+After the transaction is committed, you can read the historical data.
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply as-of-timestamp.md:178]: 这个地方的中文“即可读取最新数据”我有点疑问，这个事务事实上进行写操作了吗？如果没有的话，对于 t 表来说“最新”的数据应该还是 1，2，3 吧？那这里 select 到的是一个历史快照？
@Yisaer PTAL. Thanks!

File: as-of-timestamp.md [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply as-of-timestamp.md:242]: Same as L178.

File: read-historical-data.md [hunk 13]
```diff
@@ -4,16 +4,22 @@ summary: Learn about how TiDB reads data from history versions.
 aliases: ['/docs/dev/read-historical-data/','/docs/dev/how-to/get-started/read-historical-data/']
 ---
 
-# Read Historical Data
+# Read Historical Data Using the System Variable `tidb_snapshot`
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply read-historical-data.md:7]: Updated in https://github.com/pingcap/docs/pull/5809/commits/167871e59deee647ff99e2b3a8b87bd4b65f7f0e.

File: as-of-timestamp.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (Yisaer) [reply as-of-timestamp.md:178]: 这里想要表达的意思是，当 Stale read 事务 commit 完后，这个事务就结束了，之后新起的新事务内就能读到最新数据。

[the remark this answers is not in the document]
Comment (nolouch) [reply as-of-timestamp.md]: 这里应该是 `latest` 的，这个事务单独是一个历史读的事务。提交后用 `select * from` 可以读到最新的。

[the remark this answers is not in the document]
Comment (CharLotteiu) [reply as-of-timestamp.md]: Updated. Thanks~

**Review by TomShawn — APPROVED** [review]
LGTM

**Review by nolouch — APPROVED** [review]

**Pull request merged**